src/flb_pack.c

```diff
@@ -1058,6 +1058,259 @@ flb_sds_t flb_pack_msgpack_to_json_format(const char *data, uint64_t bytes,
     return out_buf;
 }
 
+flb_sds_t flb_pack_msgpack_to_json_format_with_trace(const char *data, uint64_t bytes,
+                                          int json_format, int date_format,
+                                          flb_sds_t date_key)
+{
```

Comment:
@tarruda I'm fine with folding the functionality into the original function. I think I might not even need the function anymore since it was used for the initial implementation where I was inlining traces inside the actual records by appending them. That approach led to too many problems, especially since most filters would just chop them off.  Let me check that out first.